lib/diagnostic.js

```diff
@@ -13,14 +26,36 @@ const min = function min() {
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
+  let numHolder = null;
+  if (arguments.length === 0) {
+    numHolder = undefined;
+  } else {
+    numHolder = arguments[0];
+  }
+
+  for (let i = 1; i < arguments.length; i++) {
+    numHolder += arguments[i];
+  }
 
+  return numHolder;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
+  let arrayCopy = [];
+  if (typeof defaultValue === 'function') {
+    for (let i = 0; i < length; i++) {
+      arrayCopy.push(defaultValue(i));
+    }
+  } else {
+    for (let i = 0; i < length; i++) {
+      arrayCopy.push(defaultValue);
+    }
+  }
 
+  return arrayCopy;
```

Comment:
so instead of having two for loops inside of the if/else it would make more sense to have one for loop around the if/else.  like this:
```
let arrayCopy = [];
for (let i = 0; i < length; i++) {
	if (typeof defaultValue === 'function') {
		arrayCopy.push(defaultValue(i));
	} else {
		arrayCopy.push(defaultValue);
	}
}
return arrayCopy;
```